src/transformers/models/bart/modeling_bart.py

```diff
@@ -750,18 +733,15 @@ def forward(
         assert v is not None
         attn_output = torch.bmm(attn_probs, v)
         assert attn_output.size() == (bsz * self.num_heads, tgt_len, self.head_dim)
-        attn_output = attn_output.transpose(0, 1).contiguous().view(tgt_len, bsz, embed_dim)
-        attn_output = self.out_proj(attn_output)
+        attn_output = (
+            attn_output.view(bsz, self.num_heads, tgt_len, self.head_dim)
+            .transpose(1, 2)
+            .reshape(bsz, tgt_len, embed_dim)
+        )
 
-        return attn_output, attn_weights_reshaped
+        attn_output = self.out_proj(attn_output)
 
-    def _concat_saved_state(self, k, v, saved_state, static_kv, bsz) -> Tuple[Tensor]:
-        # saved states are stored with shape (bsz, num_heads, seq_len, head_dim)
-        prev_K = saved_state["prev_key"].view(bsz * self.num_heads, -1, self.head_dim)
-        prev_V = saved_state["prev_value"].view(bsz * self.num_heads, -1, self.head_dim)
-        new_K = prev_K if static_kv else torch.cat([prev_K, k], dim=1)
-        new_V = prev_V if static_kv else torch.cat([prev_V, v], dim=1)
-        return new_K, new_V
+        return attn_output, attn_weights_reshaped, past_key_value
```

Comment:
`past_key_value` is usually a tuple and a stateless variable. Let's do the same for Bart.

Reply:
Should also simplify Bart + ONNX, Bart + torchscript